### ISSUE TITLE
Add the Show Header command to the command palette

### ISSUE THREAD

**Issue by jtpio** [jupyter/notebook#6401]
### Problem

The Show Header command is available as a menu item:

https://user-images.githubusercontent.com/591645/131216691-95a67ec2-7f32-416d-b627-5ebc38bb2f46.mp4

However it is not available in the command palette.

### Proposed Solution

The plugin adding the command is defined here:

https://github.com/jupyter/notebook/blob/9e81700947aec36a184bea890e09e46d37dccaf3/packages/application-extension/src/index.ts#L424-L463

We'll need to add the optional `ICommandPalette` to add the command to the palette if available.

As an example implementation, the command to toggle zen mode is added to the palette here:

https://github.com/jupyter/notebook/blob/9e81700947aec36a184bea890e09e46d37dccaf3/packages/application-extension/src/index.ts#L526-L528

**Comment by jeewonkoo** [comment]
Hello, I found this issue on https://ovio.org/projects and would love to contribute! I am new to helping Github open source but have few years of coding experience. Please let me know!

**Comment by jtpio** [comment]
Thanks @jeewonkoo!

The first step would be to fork the repository and follow the contributing guide to get the local environment up and running: https://github.com/jupyter/notebook/blob/main/CONTRIBUTING.md. Up to being able to open the application with `jupyter notebook`.

And then follow the similar code as hinted in the top comment: https://github.com/jupyter/notebook/issues/6401#issue-1217384254

But located in this plugin: https://github.com/jupyter/notebook/blob/ec4ce20b919d78f9ab5cc7fcc16d2be022aef0d0/packages/application-extension/src/index.ts#L478-L545

That would mean adding `ICommandPalette` to the list of `optional`: 

https://github.com/jupyter/notebook/blob/ec4ce20b919d78f9ab5cc7fcc16d2be022aef0d0/packages/application-extension/src/index.ts#L484

And then the code to add the command to the palette:

```ts
if (palette) { 
   palette.addItem({ command: CommandIDs.toggleTop }); 
}
```

Don't hesitate to open a pull request early, even if it's not finished or ready yet. Thanks!

**Comment by jeewonkoo** [comment]
Sounds good! I will work on it.

**Comment by jtpio** [comment]
Fixed by #6415, thanks @jeewonkoo!
